clippy_lints/src/strings.rs

```diff
@@ -208,16 +214,39 @@ fn is_string(cx: &LateContext<'_>, e: &Expr<'_>) -> bool {
     is_type_lang_item(cx, cx.typeck_results().expr_ty(e).peel_refs(), LangItem::String)
 }
 
-fn is_add(cx: &LateContext<'_>, src: &Expr<'_>, target: &Expr<'_>) -> bool {
+fn sugg_push_or_pushstr(cx: &LateContext<'_>, s: &Expr<'_>) -> String {
```

Comment:
This should take an `app: &mut Applicability`, for use with `snippet_with_applicability` instead of `snippet`.

Also, this could return an `Option<String>`, returning `None` if the string to replace is empty, so that the lint can be specialized.